test/test_heev.cc

```diff
@@ -289,30 +318,53 @@ void test_heev_work(Params& params, bool run)
+                iwork.resize(liwork);
+            }
 
             //==================================================
             // Run ScaLAPACK reference routine.
             //==================================================
             double time = barrier_get_wtime(MPI_COMM_WORLD);
-            scalapack_pheev(job2str(jobz), uplo2str(uplo), n,
-                            &Aref_data[0], 1, 1, A_desc,
-                            &Lambda_ref[0],
-                            &Z_data[0], 1, 1, Z_desc,
-                            &work[0], lwork, &rwork[0], lrwork, &info_tst);
+            if (algorithm == slate::Algorithm::EigenvalueQR) {
+                scalapack_pheev(job2str(jobz), uplo2str(uplo), n,
+                                &Aref_data[0], 1, 1, A_desc,
+                                &Lambda_ref[0],
+                                &Z_data[0], 1, 1, Z_desc,
+                                &work[0], lwork, &rwork[0], lrwork, &info_tst);
+            }
+            else if (algorithm == slate::Algorithm::EigenvalueDC) {
+                scalapack_pheevd(job2str(jobz), uplo2str(uplo), n,
+                                &Aref_data[0], 1, 1, A_desc,
+                                &Lambda_ref[0],
+                                &Z_data[0], 1, 1, Z_desc,
+                                &work[0], lwork, &rwork[0], lrwork,
+                                &iwork[0], liwork, &info_tst);
+            }
             slate_assert(info_tst == 0);
             time = barrier_get_wtime(MPI_COMM_WORLD) - time;
 
             params.ref_time() = time;
 
-            // Reference Scalapack was run, check reference against test
-            // Perform a local operation to get differences Lambda = Lambda - Lambda_ref
-            blas::axpy( n, -1.0, &Lambda_ref[0], 1, &Lambda[0], 1 );
+            // todo: do we still need this?
+            // Reset omp thread number
+            //slate_set_num_blas_threads(saved_num_threads);
 
-            // Relative forward error: || Lambda_ref - Lambda || / || Lambda_ref ||.
-            params.error() = blas::asum( n, &Lambda[0], 1 )
-                           / blas::asum( n, &Lambda_ref[0], 1 );
+            if (! ref_only) {
+                // Reference Scalapack was run, check reference against test
+                // Perform a local operation to get differences Lambda = Lambda - Lambda_ref
+                blas::axpy( n, -1.0, &Lambda_ref[0], 1, &Lambda[0], 1 );
 
-            params.okay() = params.okay() && (params.error() <= tol);
+                // Relative forward error: || Lambda_ref - Lambda || / || Lambda_ref ||.
+                params.error() = blas::asum( n, &Lambda[0], 1 )
+                    / blas::asum( n, &Lambda_ref[0], 1 );
+
+                params.okay() = params.okay() && (params.error() <= tol);
+            }
+            else {
+                params.okay() = -1; // no check since slate was not run, only ref was run
```

Comment:
I think this `else` clause isn't needed. It seems other testers don't have it. If `okay` isn't assigned, its default value is already -1 (no check).